controllers/queue_controller.go

```diff
@@ -47,11 +47,30 @@ func (r *QueueReconciler) DeleteFunc(ctx context.Context, client rabbitmqclient.
 	logger.Info("Deleting queues from ReconcilerFunc DeleteObj")
 
 	queue := obj.(*topology.Queue)
-	err := validateResponseForDeletion(client.DeleteQueue(queue.Spec.Vhost, queue.Spec.Name))
-	if errors.Is(err, NotFound) {
+	queueDeleteOptions, err := internal.GenerateQueueDeleteOptions(queue)
+	if err != nil {
+		return fmt.Errorf("failed to generate queue delete options: %w", err)
+	}
+	// Manage Quorum queue deletion if DeleteIfEmpty or DeleteIfUnused is true
+	if queue.Spec.Type == "quorum" && (queue.Spec.DeleteIfEmpty || queue.Spec.DeleteIfUnused) {
+		qInfo, err := client.GetQueue(queue.Spec.Vhost, queue.Spec.Name)
+		if err == nil {
+			return fmt.Errorf("cannot get %w Queue Information to verify queue is empty/unused: %s", err, queue.Spec.Name)
+		}
+		if qInfo.Messages > 0 && queue.Spec.DeleteIfEmpty {
+			return fmt.Errorf("Cannot delete queue %s because messages", queue.Spec.Name)
```

Comment:
Error messages should not be capitalised. This would read better as `cannot delete queue %s because it has ready messages`